### PR TITLE
Simplify erc20 contract: Part 1

### PR DESCRIPTION
**Pull request by ethanfrey**
ERC20 inspired many helpers in `cw_storage` (PrefixedStorage was basically copied from there), and error helpers in cosmwasm. Let's import these and use them to simplify the code (and reduce line count).

I also moved the files to the top level, but kept `contract.rs` and `test.rs` separate.

This is byte-wise compatible with the current contract, just cleaned up using newer helper functions. I will make another PR with a deeper change in Storage to use more helpers, but which will change the raw disk storage (but not the public APIs)

### CODE REVIEW

File: erc20/Cargo.toml [hunk 1]
```diff
@@ -30,8 +30,11 @@ cranelift = [ "cosmwasm-vm/default-cranelift"]
 singlepass = [ "cosmwasm-vm/default-singlepass"]
 
 [dependencies]
-cosmwasm = { version = "0.6.3" }
+cosmwasm = { version = "0.6.4" }
+cw-storage = "0.1.1"
 schemars = "=0.5"
+named_type = "=0.2.1"
+named_type_derive = "=0.2.1"
```
Comment (webmaster128) [inline erc20/Cargo.toml]: Why those version fixes with `=`?  Don't we want to receive patched when upgrading dependencies?

File: erc20/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,8 @@
 pub mod contract;
 
+#[cfg(test)]
+mod tests;
```
Comment (webmaster128) [inline erc20/src/lib.rs]: Ah, yeah, same effect as I originally intended but simpler file structure 👍

File: erc20/Cargo.toml (hunk 1 above)
Comment (ethanfrey) [reply erc20/Cargo.toml]: It hadn't been updated in a year, so I figured safer to pin than allow rogue upgrades.

I just checked at it was bumped to 0.2.2... but just to add a deprecation warning. Seems like this is now in rust std:

https://doc.rust-lang.org/std/any/fn.type_name.html

Comment (ethanfrey) [reply erc20/Cargo.toml]: Made an issue to fix this up: https://github.com/confio/cw-storage/issues/9

**Pull request merged**